_setuputils.py

```diff
@@ -244,6 +244,10 @@ def cythonize1(ext):
                 'infer_types': True,
                 'nonecheck': False,
             },
+            compile_time_env={
+                'PY39B1': sys.hexversion >= 0x030900B1,
+                'PY311A6': sys.hexversion >= 0x030B00A6,
+            },
```

Comment:
Please use C macros instead of Pyrex defines. Taking this kind of decision based on the runtime environment of the build is wrong, it needs to be taken at C compile time.